Description: move fallback definition of MAXPATHLEN outside the WinNT block
 helps the hurd build move a bit further
 .
 Additionally, include common.h in gpre.h to get the MAXPATHLEN define
 .
 The effor to build firebird on hurd is tracked at
 <https://bugs.debian.org/626931>
Author: Damyan Ivanov
--- a/src/gpre/gpre.h
+++ b/src/gpre/gpre.h
@@ -66,6 +66,7 @@
 #include "dyn_consts.h"
 #include "../jrd/ibase.h"
 #include "../jrd/constants.h"
+#include "../common/common.h"
 
 #ifdef GPRE_FORTRAN
 #if defined AIX || defined AIX_PPC || defined __sun
--- a/src/common/common.h
+++ b/src/common/common.h
@@ -539,14 +539,6 @@ extern "C" int remove(const char* path);
 #define API_ROUTINE_VARARG      __cdecl
 #define CLIB_ROUTINE    __cdecl
 
-#ifndef MAXPATHLEN
-#ifdef MAX_PATH
-#define MAXPATHLEN MAX_PATH
-#else
-#define MAXPATHLEN 260
-#endif
-#endif
-
 #define ERRNO		GetLastError()
 #define INET_ERRNO	WSAGetLastError()
 #define H_ERRNO		WSAGetLastError()
@@ -559,6 +551,15 @@ extern "C" int remove(const char* path);
 #endif /* WIN_NT */
 
 
+#ifndef MAXPATHLEN
+#ifdef MAX_PATH
+#define MAXPATHLEN MAX_PATH
+#else
+#define MAXPATHLEN 260
+#endif
+#endif
+
+
 #ifndef FB_CPU
 #error Define FB_CPU for your platform
 #endif
